src/main/java/seedu/address/model/person/Bloodtype.java

```diff
@@ -23,7 +23,7 @@
     public Bloodtype(String bloodtype) {
         requireNonNull(bloodtype);
         checkArgument(isValidBloodtype(bloodtype), MESSAGE_CONSTRAINTS);
-        value = bloodtype;
+        value = bloodtype.toUpperCase();
```

Comment:
Would it be better to declare a bloodtype enum?

Reply:
This should be easier since if we use enum, we have to use many if else or switch cases to change it into enum. For this we can just use it as a string

Reply:
`Enum.valueOf` can be used to generate enum. It will throw exception if the string does not match any enum value.

Reply:
Alternatively I think William's HashSet is a reasonable solution